core/arch/riscv/include/kernel/delay_arch.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#ifndef __KERNEL_DELAY_ARCH_H
+#define __KERNEL_DELAY_ARCH_H
+
+#include <platform_config.h>
+#include <riscv.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+static inline uint64_t timeout_init_us(uint32_t us)
+{
+	return read_time() + (((uint64_t)us * RISCV_MTIME_RATE) / 1000000ULL);
```

Comment:
So given my previous comment this shoud use `CFG_RISCV_MTIME_RATE`, but what if `CFG_RISCV_M_MODE` is not set?

Reply:
Didn't get it.

Reply:
Sorry...
1. `RISCV_MTIME_RATE` does not exist (assuming my [comment](https://github.com/OP-TEE/optee_os/pull/5716#discussion_r1061332653) is addressed) so that should be `CFG_RISCV_MTIME_RATE`.
2. But then I am wondering, is `CFG_RISCV_MTIME_RATE` always defined and meaningful? I am wondering this for two reasons: (a) it is currently only in `plat-spike`, what about other platforms? and (b) there is MTIME in the name which implies M mode doesn't it? so what about platforms configured for a different mode?

Reply:
Maybe the name `CFG_RISCV_MTIME_RATE ` is confusing due to the M, but that variable holds the clock frequency of `mtime` which is different across platforms and should be always defined.  The `mtime` register depends on a global real time clock. In S-Mode we read it through CLINT, in M-Mode we directly read the CSR, We can find better name otherwise.

Reply:
Ah, that makes sense. Excuse my poor knowledge of the RISC-V architecture :-/ I suppose the name is OK then.

Reply:
All comments addressed, any tag to apply to the remaining commits ?

Reply:
You should also drop the extra parentheses `+ ( ... );`. Then please apply:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

to "riscv: kernel: include: add delay_arch.h"